inspirehep/utils/tickets.py

```diff
@@ -24,26 +24,222 @@
-        tracker.login()
-        return tracker
+    return rt_instance
 
 
 def retry_if_connection_problems(exception):
     """Return True if exception is an ConnectionError in Rt."""
     return isinstance(exception, ConnectionError)
+
+
+def create_ticket(queue, requestors, subject, body, recid=None, **kwargs):
+    """Creates new RT ticket and returns new ticket id.
+
+    :param queue: queue name for the ticket
+    :type queue: string
+
+    :param requestors: username to set to requestors field of the ticket
+    :type requestors: string
+
+    :param body: message body of the ticket
+    :type body: string
+
+    :param recid: record id to be set custom RecordID field
+    :type recid: integer
+
+    :param kwargs: Other arguements possible to set:
+
+                   Cc, AdminCc, Owner, Status,Priority, InitialPriority,
+                   FinalPriority, TimeEstimated, Starts, Due,
+                   ... (according to RT fields)
+
+                   Custom fields CF.{<CustomFieldName>} could be set
+                   with keywords CF_CustomFieldName.
+    :returns: ID of the new ticket or ``-1``, if it fails
+    :rtype: integer
+    """
+    body = _strip_lines(body)
+
+    payload = dict(
+        Queue=queue,
+        Subject=subject,
+        Text=body,
+        **kwargs
+    )
+
+    if recid:
+        payload["CF_RecordID"] = str(recid)
+
+    # Check if requests is set and also ignore admin due to RT mail loop
+    if requestors and "admin@inspirehep.net" not in requestors:
+        payload["requestors"] = requestors
+
+    return rt_instance.create_ticket(**payload)
+
+
+def create_ticket_with_template(queue,
+                                requestors,
+                                template_path,
+                                template_context,
+                                recid=None,
+                                **kwargs):
+    """Creates new RT ticket with a body that is rendered template
+
+    :param queue: queue name for the ticket
+    :type queue: string
+
+    :param requestors: username to set to requestors field of the ticket
+    :type requestors: string
+
+    :param template_path: path to the template for the ticket body
+
+    :param template_context: context object to be used to render template
+
+    :param recid: record id to be set custom RecordID field
+    :type recid: integer
+
+    :param kwargs: Other arguements possible to set:
+
+                Cc, AdminCc, Owner, Status,Priority, InitialPriority,
+                FinalPriority, TimeEstimated, Starts, Due,
+                ... (according to RT fields)
+
+                Custom fields CF.{<CustomFieldName>} could be set
+                with keywords CF_CustomFieldName.
+    :returns: ID of the new ticket or ``-1``, if it fails
+    :rtype: integer
+    """
+    body = render_template(template_path, **template_context).strip()
+    subject = template_context.get("subject")
+    # FIXME: raise exception if subject is None ???
```

Comment:
it is gonna have `"None"` subject. Maybe we can also put `"no subject"` instead of raising.